File: Modules/HumanResource/Announcement/AnnouncementDepartmentBinded/AnnouncementDepartmentBindedForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.AnnouncementDepartmentBinded")]
[BasedOnRow(typeof(AnnouncementDepartmentBindedRow), CheckNames = true)]
public class AnnouncementDepartmentBindedForm
{
    public int AnnouncementWizardId { get; set; }
    public int DepartmentId { get; set; }
}

File: Modules/ServerTypes/Announcement/AnnouncementDepartmentBindedForm.ts
import { IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementDepartmentBindedForm {
    AnnouncementWizardId: IntegerEditor;
    DepartmentId: IntegerEditor;
}

export class AnnouncementDepartmentBindedForm extends PrefixedContext {
    static formKey = 'Announcement.AnnouncementDepartmentBinded';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementDepartmentBindedForm.init)  {
            AnnouncementDepartmentBindedForm.init = true;

            var w0 = IntegerEditor;

            initFormType(AnnouncementDepartmentBindedForm, [
                'AnnouncementWizardId', w0,
                'DepartmentId', w0
            ]);
        }
    }
}